pkgs/stdenv/generic/builder.sh

```diff
@@ -17,7 +19,7 @@ mkdir $out
 
 # Allow the user to install stdenv using nix-env and get the packages
 # in stdenv.
-mkdir $out/nix-support
-if [ "$propagatedUserEnvPkgs" ]; then
-    printf '%s ' $propagatedUserEnvPkgs > $out/nix-support/propagated-user-env-packages
+mkdir "${out}/nix-support"
+if [ "${propagatedUserEnvPkgs-}" ]; then
+    printf '%s ' "$propagatedUserEnvPkgs" > "${out}/nix-support/propagated-user-env-packages"
 fi
```

Comment:
```suggestion
mkdir "$out/nix-support"
if [ -n "${propagatedUserEnvPkgs-}" ]; then
    printf '%s ' "$propagatedUserEnvPkgs" > "$out/nix-support/propagated-user-env-packages"
fi
```

I think we should try to make as little diff as possible. From my impression most shell scripts omit the optional brackets around the dollar sign and I personally think we should continue to do that. I also think it makes the code easier to understand in most situations by having less special characters to interpret.

---

The -n suggestion comes from https://github.com/koalaman/shellcheck/wiki/SC2244 which is disabled by default. I have it enabled in my setup and think it would be quite nice to follow.

Reply:
I don't think we should omit the braces, because they are only optional in the presence of certain characters. If the characters after the variable change, that completely changes the semantics of the interpolation.

Weird, if I run `shellcheck --include=SC2244 pkgs/stdenv/generic/builder.sh` I don't see any error about a missing `-n`. But I agree that's better, so I've added it.

Reply:
> I don't think we should omit the braces

We don't have any style guide about this. Please avoid changing such opinionated things around. I am pretty sure shellcheck does not warn about those.

> because they are only optional in the presence of certain characters. If the characters after the variable change, that completely changes the semantics of the interpolation.

All the editors I use have syntax highlighting and tell me immediately when such an edge case occurs and I need to add the extra ``{}`` but that happens so rarely that I don't think it justifies adding them as a precaution everywhere. 

> Weird, if I run shellcheck --include=SC2244 pkgs/stdenv/generic/builder.sh I don't see any error about a missing -n. But I agree that's better, so I've added it.

I think you need to pass in ``--enable=avoid-nullary-conditions`` or it needs to be written to ``~/.shellcheckrc``.

Reply:
I'd be interested to see what other community members say about this syntax. I wouldn't be surprised if others use editors without syntax highlighting, or with broken syntax highlighting. I've yet to see a single IDE which highlights advanced Bash syntax properly, so I tend to err on the side of "let's make bugs as glaringly obvious as possible."